core/variant/typed_array.h

```diff
@@ -40,19 +40,29 @@
 
 template <class T>
 class TypedArray : public Array {
+	static_assert(is_variant_type_v<T>, "Only Variant types are allowed for typed arrays.");
+
+	static String _class_name() {
```

Comment:
I was immitating the private fields in `Array`, which are prefixed with underscores. Didn't give it much thought beyond that and wouldn't be opposed to renaming it, but the two below constructors should be the only time it's ever used.

Reply:
Part of me is questioning if this separate function and the early `""` return is even necessary. I do not have the expertise to say but what does "catching" this as early as possible do?

Reply:
My concern is about performance here since this is a pretty "critical" part of the engine being accessed here. By all means it probably does not matter, but it's always on the back of my head.
I do like this restructuring myself but this is the only part of this PR that I find strange.

Reply:
As I understand it, making these evaluations `constexpr` would be strictly beneficial for performance, as everything gets evaluated at compile-time. It's possible that the compilation time itself takes a hit, but I haven't observed anything suggesting that to be the case. But I'm also not an expert on how this stuff works under the hood, so take this with a grain of salt.

In any case, I've updated the PR to add the new `type_info` properties so those evaluations are isolated and can be reused anywhere. Already put into action by having the same function be used in the constructors and `GetTypeInfo`.